### PR TITLE
user _limitZoom in flyTo, like we do in resetView

### PR DESCRIPTION
**Pull request by yohanboniface**
Thanks :)

### CODE REVIEW

File: spec/suites/map/MapSpec.js [hunk 1]
```diff
@@ -1418,6 +1418,24 @@ describe('Map', () => {
 
 			map.flyTo(dc, 4, {duration: 0.1});
 		});
+
+		it('flyTo should honour maxZoom', function (done) {
+			this.timeout(10000); // This test takes longer than usual due to frames
```
Comment (Falke-Design) [inline spec/suites/map/MapSpec.js:1423]: Can we speed up the test with setting `animate: false`? `map.flyTo(newCenter, 22, {animate:false});`

**Review by Falke-Design — CHANGES_REQUESTED** [review]

File: spec/suites/map/MapSpec.js (hunk 1 above)
Comment (yohanboniface) [reply spec/suites/map/MapSpec.js:1423]: I'd say that doing so, we'll not pass in the animation code, which is where the unbounded zoom was. :/

Comment (Falke-Design) [reply spec/suites/map/MapSpec.js:1423]: Yes that is true, sorry I didn't saw that. But please change the duration and remove the timeout. `map.flyTo(newCenter, 22, {animate: true, duration: 0.1});`

Comment (yohanboniface) [reply spec/suites/map/MapSpec.js]: Done! Thanks for the suggestion :)

File: spec/suites/map/MapSpec.js [hunk 2]
```diff
@@ -1418,6 +1418,21 @@ describe('Map', () => {
 
 			map.flyTo(dc, 4, {duration: 0.1});
 		});
+
+		it('flyTo should honour maxZoom', () => {
```
Comment (Falke-Design) [inline spec/suites/map/MapSpec.js:1422]:
```suggestion
		it('flyTo should honour maxZoom', (done) => {
```

File: spec/suites/map/MapSpec.js [hunk 3]
```diff
@@ -1418,6 +1418,21 @@ describe('Map', () => {
...
+			const newCenter = latLng(10, 11),
+			    maxZoom = 20;
+			map.options.maxZoom = maxZoom;
+
+			map.setView([0, 0], 0);
+
+			map.on('zoomend', () => {
+				expect(map.getCenter()).to.eql(newCenter);
+				expect(map.getZoom()).to.eql(maxZoom);
```
Comment (Falke-Design) [inline spec/suites/map/MapSpec.js:1431]:
```suggestion
				expect(map.getZoom()).to.eql(maxZoom);
				done();
```

**Review by Falke-Design — CHANGES_REQUESTED** [review]
We need the `done` because else if `zoomend` is never called, it would be still successfull

**Comment by yohanboniface** [comment]
> We need the `done` because else if `zoomend` is never called, it would be still successfull

Ah, damn, you're right!

**Review by Falke-Design — APPROVED** [review]
Looks good now, thank you!

**Pull request merged**